### PR TITLE
Register pipeline with funcx

### PR DESCRIPTION
**Pull request by WillEngler**
Implements #51

## Overview

This PR implements the `garden-ai pipeline register` command.

## Discussion

Some little refactoring notes:
- Our `utils.py` is getting to be too big of a grab bag imo and we should start splitting it. I made it into a directory but didn't yet divvy up the helpers in the original file. Let's do that and then get rid of `utils/misc.py`
- I've instructed black, mypy, and flake8 to ignore the `fixtures` directory so that there can be broken Python in there for testing purposes.

## Testing

I've run the `register` command from the integration test included in this PR. Admittedly, the real test of whether the registration worked will be if we can invoke it!

----
:books: Documentation preview :books:: https://garden-ai--90.org.readthedocs.build/en/90/

### CODE REVIEW

File: garden_ai/utils/filesystem.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    """
+
+    module_name = python_file.stem
+    spec = importlib.util.spec_from_file_location(module_name, python_file)
+
+    if spec is None or spec.loader is None:
+        raise PipelineLoadException("Could not load the module from file")
+
+    module = importlib.util.module_from_spec(spec)
+    sys.modules[module_name] = module
```
[the remark this answers is not in the document]
Comment (OwenPriceSkelly) [reply garden_ai/utils/filesystem.py:34]: "more of a question than a comment, really..."

File: tests/fixtures/fixture_pipeline/invalid_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+## THIS FILE WAS AUTOMATICALLY GENERATED ##
+from garden_ai import GardenClient, Model, Pipeline, step
+
+Here is some text that is not valid Python. Oops
```
Comment (OwenPriceSkelly) [inline tests/fixtures/fixture_pipeline/invalid_pipeline.py:4]: 😤

File: garden_ai/client.py [hunk 3]
```diff
@@ -316,6 +315,12 @@ def build_container(self, pipeline: Pipeline) -> str:
         built_container_uuid = build_container(self.funcx_client, pipeline)
         return built_container_uuid
 
+    def register_pipeline(self, pipeline: Pipeline, container_uuid: str) -> str:
+        func_uuid = register_pipeline(self.funcx_client, pipeline, container_uuid)
+        pipeline.funcx_uuid = UUID(func_uuid)
```
Comment (OwenPriceSkelly) [inline garden_ai/client.py:320]: now that we can do it, shall we DOI it?

Comment (OwenPriceSkelly) [reply garden_ai/client.py:320]:
```suggestion
        pipeline.funcx_uuid = UUID(func_uuid)
        self._mint_doi(pipeline)
```

File: tests/integrations/test_pipeline_registration.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import pytest
+
+from garden_ai.app.pipeline import register
```
Comment (OwenPriceSkelly) [inline tests/integrations/test_pipeline_registration.py:3]: I think (but don't totally remember why) it's better to test the cli commands with the `typer.testing.CliRunner` class like in the `tests/test_cli.py` than to exercise the underlying functions directly

File: .github/workflows/ci.yaml
Comment (OwenPriceSkelly) [inline .github/workflows/ci.yaml:1]: while we're at it, I think we can add 3.11 back into the mix now that the numba issue upstream is fixed

**Review by OwenPriceSkelly — CHANGES_REQUESTED** [review]
nothing major! just picking nits

File: .github/workflows/ci.yaml
Comment (WillEngler) [reply .github/workflows/ci.yaml:1]: Noted! I think I'll punt on lumping that in to this PR but let's take advantage of that soon

File: garden_ai/client.py (hunk 3 above)
Comment (WillEngler) [reply garden_ai/client.py:320]: Oh yeah! Thanks for the reminder, this is definitely required behavior for pipeline registration.

File: garden_ai/utils/filesystem.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+from garden_ai import Pipeline
+
+
+class PipelineLoadException(Exception):
+    """Exception raised when a container build request fails"""
+
+    pass
+
+
+def load_pipeline_from_python_file(python_file: Path, pipeline_name: str) -> Pipeline:
```
[the remark this answers is not in the document]
Comment (WillEngler) [reply garden_ai/utils/filesystem.py:13]: Love this. I don't want to make the user do unnecessary bookkeeping so this is a big improvement to the UX

**Review by OwenPriceSkelly — APPROVED** [review]
LGTM!

**Pull request merged**